delivery_tnt_oca/models/delivery_carrier.py

```diff
@@ -0,0 +1,170 @@
+    tnt_oca_ws_password = fields.Char(string="ExpressConnect Password WS")
+    tnt_oca_ws_account = fields.Char(string="ExpressConnect Account WS")
+    # Misc
+    tnt_product_type = fields.Selection(
+        selection=[
+            ("D", "Document (paper/manuals/reports)"),
+            ("N", "Non-document (packages)"),
+        ],
+        default="N",
+        string="Product type",
+    )
+    tnt_product_code = fields.Char(
+        compute="_compute_tnt_product_code", string="Product code",
+    )
+    tnt_product_code_d = fields.Selection(
+        selection=[
+            ("09D", "9:00 EXPRESS"),
+            ("12D", "12:00 EXPRESS"),
+            ("15D", "GLOBAL EXPRESS"),
+        ],
+        default="09D",
+        string="Product code (Docs)",
+    )
+    tnt_product_code_n = fields.Selection(
+        selection=[("15N", "GLOBAL EXPRESS"), ("48N", "ECONOMY EXPRESS")],
+        default="15N",
+        string="Product code (Non docs)",
+    )
+    tnt_product_service = fields.Char(
+        compute="_compute_tnt_product_service", string="Product service",
+    )
+    tnt_product_service_d = fields.Selection(
+        selection=[
+            ("EX", "Express"),
+            ("EX09", "9:00 Express"),
+            ("EX10", "10:00 Express"),
+            ("EX12", "12:00 Express"),
+        ],
+        default="EX",
+        string="Product service (Docs)",
+    )
+    tnt_product_service_n = fields.Selection(
+        selection=[
+            ("EC", "Economy Express"),
+            ("EC12", "12:00 Economy Express"),
+            ("EX", "Express"),
+            ("EX09", "9:00 Express"),
+            ("EX10", "10:00 Express"),
+            ("EX12", "12:00 Express"),
+        ],
+        default="EX",
+        string="Product service (Non docs)",
+    )
+    tnt_payment_indicator = fields.Selection(
+        selection=[("S", "Sender pays"), ("R", "Receiver pays")],
+        default="S",
+        string="Payment indicator",
+    )
+    tnt_line_of_business = fields.Selection(
+        selection=[
+            ("1", "Domestic dk to dk transfers"),
+            ("2", "international non-domestic transfers"),
+        ],
+        default="1",
+        string="Line of business",
+    )
+    tnt_collect_time_from = fields.Char(default="10:30", string="Collect time from",)
+    tnt_collect_time_to = fields.Char(default="16:00", string="Collect time to",)
+
+    def _compute_tnt_product_code(self):
+        for item in self:
+            if item.tnt_product_type == "D":
+                item.tnt_product_code = item.tnt_product_code_d
+            else:
+                item.tnt_product_code = item.tnt_product_code_n
+
+    def _compute_tnt_product_service(self):
+        for item in self:
+            if item.tnt_product_type == "D":
+                item.tnt_product_service = item.tnt_product_service_d
+            else:
+                item.tnt_product_service = item.tnt_product_service_n
+
+    def tnt_oca_rate_shipment(self, order):
```

Comment:
This doesn't seem like the kind of method that should be called from RPC, right? So better make it private:

```suggestion
    def _tnt_oca_rate_shipment(self, order):
```

Reply:
This method cannot be changed (the function name is defined at https://github.com/odoo/odoo/blob/13.0/addons/delivery/models/delivery_carrier.py#L149).